sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/memory.scala

```diff
@@ -110,6 +117,7 @@ class MemorySink(schema: StructType) extends Sink with Logging {
   }
 
   override def addBatch(nextBatch: Batch): Unit = synchronized {
+    nextBatch.data.collect()  // 'compute' the batch's data and record the batch
```

Comment:
This is to force the memory sink to compute the batch and throw error if the computation is supposed to.